### DISCUSSION TITLE
How to read EAN-13 code with insufficient quiet zone?

### DISCUSSION THREAD

**Discussion by kientux** [zxing-cpp/zxing-cpp#509] [Q&A] [answered]
I'm trying to read this barcode:

![barcode-large](https://user-images.githubusercontent.com/4200743/216565472-7ed36cc1-d672-4e28-a1c6-04feb8b8952d.jpeg)

I figured out that zxing-cpp can't read this because of the black-line box around the barcode. Actually it's because of the right side of the box. If I remove that, zxing-cpp read the barcode instantly.

![barcode-large-edited](https://user-images.githubusercontent.com/4200743/216565511-9442fcc6-b5ed-48e4-a4cb-50c253c75d71.jpeg)

If I crop the image into this:

![barcode-large](https://user-images.githubusercontent.com/4200743/216572073-08e97b90-38a2-423b-95d2-16f2bc5ebbc0.jpeg)

then zxing-cpp can reads but still very hard (tried multiple times/positions of camera). Read directly from image still can't work.

So is there anyway for zxing-cpp to improve reading in this case? Before use zxing-cpp, I used another port of zxing (ZXingObjC) and it works fine. I don't know much about c++ and barcodes, but will try to help if I can, thanks.

**Comment by axxel** [comment, marked as answer]
The problem is the insufficient quiet zone on the right side of the symbol. The standard requires 7 modules, the library enforces at least 6 and this symbol has more like 4. So this is technically an invalid EAN-13 symbol.

See [here](https://github.com/zxing-cpp/zxing-cpp/blob/f25c1807558299789a0b3fa20101960cdd13e20c/core/src/oned/ODMultiUPCEANReader.cpp#L31). Reducing this value is possible but could lead to more false positives. If I reduced the 6 to 4, it still fails to decode, though. If you set the minLineCount to 1, it reads a value but the value is wrong. If I scan this with my phone off the monitor, it ready just fine, due to some increased resolution, I believe.

Note: if you scan this with your phone off the monitor and rotate the phone slightly, then the success rate increases because scanning slightly diagonally makes the quiet zone "virtually" larger.

Bottomline: I don't have a solution for you that makes this image read successfully as is.
